runtime/src/utils/buffer/write.rs

```diff
@@ -164,9 +181,26 @@ impl<B: Blob> Write<B> {
     /// This represents the total size of data that would be present after flushing.
     #[allow(clippy::len_without_is_empty)]
     pub async fn size(&self) -> u64 {
-        let inner = self.inner.read().await;
+        let buffer = self.buffer.read().await;
+        buffer.size()
+    }
 
-        inner.size()
+    /// Consumes the [Write] and returns the underlying blob.
+    ///
+    /// # Warning
+    ///
+    /// Any buffered data that hasn't been flushed or synced will be discarded.
+    pub fn take_blob(self) -> B {
```

Comment:
for consistency, I think we'd call these `take_inner` and `clone_inner` but acknowledge they don't have the same ring to them.